### PR TITLE
feat(result): show_duration duration + token usage as grey footer

### PR DESCRIPTION
**Pull request by timqi**
## Capability
When `show_duration` is enabled, the terminal result now surfaces **reasoning duration + context-window token usage** as a single de-emphasized grey subtext footnote, instead of a prominent head line in the answer body.

## What changed
- `format_result_footer()` (new) builds the compact one-line footnote: status emoji + `⏱️ time` + `🪙 tokens`, e.g. `✅ ⏱️ 2m 24s · 🪙 240k tok`. The outcome word (`Success`/`Done`) is replaced by an emoji marker (`✅`/`⚠️`/`❌`).
- The footnote rides the existing **platform subtext channel** (Slack context block, Discord `-#`, Feishu grey font) — the same styling as the concise status footer — rather than being prepended to the body.
- In concise mode it supersedes the status bubble's own terminal `✅ done · …` footer so the outcome line stays consistent (one grey line, no head text).
- Token figure reuses the concise footer source (`_session_token_total` / `_format_token_count`) via new `controller.session_token_field()`.
- `show_duration` off: behavior unchanged (no footnote head; concise footer still shows tokens).

## Behavior notes / limitations
- Telegram ignores subtext (per adapter contract) → footnote not rendered there; no error.
- Empty answer + only-timing edge case settles silently; the concise bubble collapse still shows its own footer.

## Evidence layers
- **Unit**: `tests/test_result_message_token_field.py` (footer formatting: emoji markers, time/token glyphs, omission rules, legacy `format_result_message` body path).
- **Contract/regression**: existing `test_message_dispatcher_status_bubble.py`, `test_message_dispatcher_result_fallback.py`, `test_agent_silent_result.py` pass (103 total).
- **Scenario**: no scenario catalog covers this footer; none added.
- **Residual manual**: verify grey rendering across Slack / Discord / Feishu in Incus regression.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: core/message_dispatcher.py [hunk 1]
```diff
@@ -1397,6 +1410,13 @@ async def emit_agent_message(
                     _, done_footer = self._compose_status_message(
                         context, status_consolidated_key, done=True, reason=terminal_reason, result_body=display_text
                     )
+                # A caller-supplied ``result_footer`` (the show_duration duration +
+                # token footnote) takes precedence: it renders in EVERY delivery
+                # mode as the de-emphasized subtext footer, and in concise mode it
+                # supersedes the bubble's own ``✅ done · …`` line so the terminal
+                # footer stays consistent (grey, one line, no head text).
+                if result_footer:
+                    done_footer = result_footer
```
[the remark this answers is not in the document]
Comment (timqi) [reply core/message_dispatcher.py]: Fixed in `a1acb3f4`. Root cause: setting `result_footer` made `done_footer` non-null for **every** platform, and non-subtext adapters (`telegram.send_markdown_message`, `avibe.send_message`/`send_message_with_buttons`) don't accept the `subtext` kwarg → markdown/Web-Chat results with `show_duration` on failed delivery (and Avibe got no stream chunk).

Fix — gate at the dispatcher on the existing `supports_status_bubble` capability:
- Slack/Discord/Lark (subtext-capable) → footnote rides `subtext` as before.
- Telegram/WeChat/Avibe → footnote folds onto the body (`\n\n{footer}`), so it stays visible and no unsupported `subtext` kwarg is passed.

Also added `subtext` (accept-and-ignore) to the three flagged adapter methods for BaseIMClient contract parity, plus two regression tests in `test_message_dispatcher_result_fallback.py` (subtext path vs body-fold path — the Telegram one reproduces this crash pre-fix). 👍

**Comment by timqi** [comment]
@codex review

File: core/message_dispatcher.py [hunk 2]
```diff
@@ -1397,6 +1410,24 @@ async def emit_agent_message(
...
+                #     consistent (grey, one line, no head text);
+                #   * platforms WITHOUT subtext rendering (Telegram/WeChat/Avibe)
+                #     fold it onto the body as a trailing line, so the footnote
+                #     stays visible AND their senders (which don't accept the
+                #     ``subtext`` kwarg) are never handed it.
+                if result_footer:
+                    if self._capabilities(context).supports_status_bubble:
+                        done_footer = result_footer
+                    elif display_text.strip():
+                        display_text = f"{display_text}\n\n{result_footer}"
```
[the remark this answers is not in the document]
Comment (timqi) [reply core/message_dispatcher.py]: Fixed in `70be695f`. You're right — for non-subtext platforms (Telegram/WeChat/Avibe) the footnote was folded onto the delivered body but `persist_text`/`avibe_text` were computed from the pre-fold text, so a reloaded transcript/inbox dropped it.

Fix: a shared `_fold_footer(text, footer)` helper now applies the same `\n\n{footer}` fold to both the delivered `display_text` AND the persisted text (`persist_text` for IM, `avibe_text` for Web Chat), restoring the persist-what-was-shown invariant. Subtext platforms are unaffected (footer rides `subtext`, never persisted, matching the concise done-footer). Added a regression test asserting the folded footer is persisted for Telegram.

**Comment by timqi** [comment]
@codex review

File: core/message_dispatcher.py [hunk 3]
```diff
@@ -1397,6 +1420,27 @@ async def emit_agent_message(
...
+                #     grey subtext — and in concise mode it supersedes the bubble's
+                #     own ``✅ done · …`` line so the terminal footer stays
+                #     consistent (grey, one line, no head text);
+                #   * platforms WITHOUT subtext rendering (Telegram/WeChat/Avibe)
+                #     fold it onto the body as a trailing line, so the footnote
+                #     stays visible AND their senders (which don't accept the
+                #     ``subtext`` kwarg) are never handed it.
+                folded_footer: Optional[str] = None
+                if result_footer:
+                    if self._capabilities(context).supports_status_bubble:
```
[the remark this answers is not in the document]
Comment (timqi) [reply core/message_dispatcher.py]: Fixed in `bbd4077d`. The fold-vs-subtext decision now gates on `target_context` capabilities (the actual delivery target), so a delivery_override redirecting a Slack/Discord/Lark turn to a non-subtext target folds the footer onto the body instead of handing `subtext` to an adapter that ignores it. Added a regression test with a `delivery_override` to Telegram.

File: modules/agents/base.py [hunk 4]
```diff
@@ -504,15 +504,37 @@ async def emit_result_message(
...
-            await self.controller.emit_agent_message(context, "result", formatted, parse_mode=parse_mode, is_error=is_error)
+                parts.append(visible_suffix)
+            body = "\n".join(parts)
+            await self.controller.emit_agent_message(
+                context,
+                "result",
+                body,
+                parse_mode=parse_mode,
+                is_error=is_error,
+                result_footer=result_footer,
```
[the remark this answers is not in the document]
Comment (timqi) [reply modules/agents/base.py:543]: Fixed in `bbd4077d`. A footer-only completion (show_duration on, no visible result/suffix) now promotes the footnote to the visible body before dispatch, so a duration/token-only turn is delivered/persisted/streamed instead of settling silently. Added a regression test asserting the footer becomes the visible body (and no separate subtext footer is passed).

**Comment by timqi** [comment]
@codex review

File: core/message_dispatcher.py [hunk 5]
```diff
@@ -1397,6 +1420,31 @@ async def emit_agent_message(
...
+                #     stays visible AND their senders (which don't accept the
+                #     ``subtext`` kwarg) are never handed it.
+                folded_footer: Optional[str] = None
+                if result_footer:
+                    # Gate on the DELIVERY TARGET's capabilities, not the source
+                    # context: a delivery_override can redirect a Slack/Discord/Lark
+                    # turn to a non-subtext target (or vice versa), and the footer
+                    # must follow where it is actually delivered/persisted.
+                    if self._capabilities(target_context).supports_status_bubble:
+                        done_footer = result_footer
```
[the remark this answers is not in the document]
Comment (timqi) [reply core/message_dispatcher.py:1452]: Fixed in `b1c8b5a3`. Slack's native markdown sender drops `subtext` when it falls back to the legacy mrkdwn path for bodies over `_SLACK_MARKDOWN_TEXT_LIMIT` — so I now fold `subtext` onto the body in that specific fallback branch, retaining the footnote for 12k–30k inline results. Added `test_slack_markdown_footer.py` covering the fold (and the no-subtext passthrough).

File: modules/agents/base.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (timqi) [reply modules/agents/base.py:543]: This was already addressed in `bbd4077d` (footer-only results are promoted to the visible body before dispatch) and is covered by `test_footer_only_result_is_promoted_to_visible_body` — re-surfaced here only because the diff lines shifted. No further change needed.

**Comment by timqi** [comment]
@codex review

File: core/message_dispatcher.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (timqi) [reply core/message_dispatcher.py:1452]: Fixed in `5c784b1c`. `folded_footer` is now set for both platform kinds, so the footnote is folded into the stored text (`persist_agent_message` + the agent-run terminal record) on Slack/Discord/Lark as well — while delivery still rides the grey `subtext` on those platforms. `persist_text` is never mutated, so there's no double-fold, and it's a no-op when no footer is present. Added a regression test asserting the subtext-platform result persists body+footer.

**Comment by timqi** [comment]
@codex review

File: core/message_dispatcher.py [hunk 6]
```diff
@@ -1397,6 +1420,34 @@ async def emit_agent_message(
...
+                #     consistent (grey, one line, no head text);
+                #   * platforms WITHOUT subtext rendering (Telegram/WeChat/Avibe)
+                #     fold it onto the body as a trailing line, so the footnote
+                #     stays visible AND their senders (which don't accept the
+                #     ``subtext`` kwarg) are never handed it.
+                # ``folded_footer`` is the footnote to APPEND to the stored text so
+                # a reloaded transcript / inbox / agent-run record keeps the
+                # duration/token info (it used to live in the result body). It is
+                # set for BOTH platform kinds; only the DELIVERY form differs.
+                folded_footer: Optional[str] = result_footer or None
```
[the remark this answers is not in the document]
Comment (timqi) [reply core/message_dispatcher.py:1443]: Fixed in the latest commit. The suppressed-delivery branch now folds `result_footer` into the recorded text (`_record_suppressed_agent_run_terminal_result` / `_record_suppressed_run_message`) since subtext isn't an option for private runs — so the stored result keeps the duration/token info. No-op when there's no footer. Added a regression test.

**Comment by timqi** [comment]
@codex review

File: modules/im/slack.py [hunk 7]
```diff
@@ -857,20 +857,21 @@ async def send_markdown_message(
...
-            # do NOT attach the done-footer subtext. Passing subtext routes the
-            # send through the status-bubble path, which rejects bodies > the cap
-            # (so a 12k–30k inline result would otherwise fail to post). The footer
-            # is a non-essential trailer; reliable delivery of the body wins.
+            # Over the markdown-block cap: keep the LEGACY mrkdwn/section path, which
+            # cannot carry a separate context-footer block (routing through the
+            # status-bubble path would reject bodies > the cap). Fold ``subtext``
+            # onto the body instead so the show_duration/token footnote is retained
+            # rather than silently dropped for 12k–30k inline results.
+            body = f"{text}\n\n{subtext}" if subtext else text
```
[the remark this answers is not in the document]
Comment (timqi) [reply modules/im/slack.py:865]: Fixed in `13a3e3c4`. The `invalid_blocks`/`unsupported_block_type` recovery branch now folds `subtext` onto the body (plain mrkdwn — not a rebuilt block, so it won't re-hit the same rejection), retaining the show_duration/token footnote on that path too. Added a regression test that simulates a block-rejection SlackApiError and asserts the folded body.

**Comment by timqi** [comment]
@codex review
